src/bindfs.c

```diff
@@ -249,8 +249,8 @@ static size_t round_up_buffer_size_for_direct_io(size_t size);
 
 static unsigned int permset_to_bits(acl_permset_t permset);
 static int access_check(const char *real_path, int wants);
```

Comment:
I believe this can return `bool` as well.

Reply:
The only reason it doesn't is because of the potential [lstat](https://github.com/vimist/bindfs/blob/acl_support/src/bindfs.c#L831) error. We're loosing that information in the `path_has_search_perms` and `dirname_access_check` functions now though, so I can make this return a boolean as well if you'd prefer?

Now I recall why I had those  functions returning an `int` before, it was to preserve that error number.

Reply:
Ah, I missed that and went by the comments that only mentioned `0` and `-EACCES`.

Let's use `errno` i.e. let's make the functions guarantee that if they return `false`, then `errno` is set, either to  (positive) `EACCES` by your checks or to whatever `lstat` sets it. Then the callsites would look like:

```c
if (!some_access_check_function(...))
    return -errno;
```

which is similar to many other calls in bindfs.